tests/views/test_switch.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+from hmpy.views.switch import SwitchView
+from PyQt5.QtCore import Qt
+
+
+def test_on_init(qtbot):
+    # Test that the value is initialized properly
+    switch = SwitchView(False)
+    qtbot.addWidget(switch)
+    assert not switch.on
+
+
+def test_on_set(qtbot):
+    # Test that the value is set properly
+    switch = SwitchView()
+    switch.on = True
```

Comment:
Should test that toggle is triggered only on change, once other comments are addressed